crates/common/tedge_utils/src/file.rs

```diff
@@ -380,7 +383,7 @@ pub fn change_user_and_group(file: &Path, user: &str, group: &str) -> Result<(),
     let gid = get_metadata(Path::new(file))?.gid();
 
     // if user and group are same as existing, then do not change
-    if (ud != uid) && (gd != gid) {
+    if (ud != uid) || (gd != gid) {
```

Comment:
This looked like a bug as I think we want to `chown` if there is any change, not just if both the `uid` and `gid` have changed.